psignifit/psignifit.py

```diff
@@ -178,11 +186,8 @@ def _fit_parameters(data: np.ndarray, bounds: ParameterBounds,
     # do first sparse grid posterior_grid evaluation
     grid = get_grid(bounds, steps_moving_bounds)
     posteriors_sparse, grid_max = posterior_grid(data, sigmoid=sigmoid, priors=priors, grid=grid)
-    # normalize the posterior_grid
-    posterior_volumes = posteriors_sparse * integral_weights([grid_value for _, grid_value in sorted(grid.items())])
-    posterior_integral = posterior_volumes.sum()
     # indices on the grid of the volumes that contribute more than `tol` to the overall integral
-    mask = np.nonzero(posterior_volumes / posterior_integral >= max_bound_value)
+    mask = np.nonzero(posteriors_sparse >= max_bound_value)
```

Comment:
This needs to be refactored.

  - document what `max_bound_value` is
  - factor the loop out to a different function called `tighten_bounds`, which should be documented properly

Reply:
Once working on this, I preferred another solution: I called the new variable thight_bounds, but the function mask_bounds.
The function is relatively general, has nothing to do with tight or loose, just setting bounds around a mask.